### PR TITLE
feat(ps): Configure scanner from a USB drive ballot package

### PR DESCRIPTION
**Pull request by carolinemodic**
Creates the main workflow for showing an unconfigured screen, and configuring from a ballot package on a usb drive when a usb with the expected ballot package is inserted.

Home screen with unconfigure is just a placeholder for testing. 

Succesfull Configuration: 

https://user-images.githubusercontent.com/14897017/116759629-a1055780-a9c7-11eb-8302-08dd118e8152.mov

When a USB without the ballot package is inserted: 

https://user-images.githubusercontent.com/14897017/116759612-95199580-a9c7-11eb-890d-c02f36467b43.mov

There are a lot of files copied over from bsd in this PR, many/all of these should eventually live in a shared utils module but since we don't have that yet I moved them in, the only modifications made were fleshing out the tests a bit more in some cases or adding istanbul skips for test coverage to pass. 

There are also a lot of types taken over from bsd. I wanted to put these in @votingworks/types but ran into an issue where they import types from hmpb-interpreter and if I do that from within /types there is a circular dependency. I'm not really sure the best place for these types to live so to unblock myself I just copied them in but I think we should follow up with moving them to some shared location.

### CODE REVIEW

**Comment by carolinemodic** [comment]
I'm confused about the bmd circleci error and how it is related to this change...

File: apps/precinct-scanner/src/AppRoot.tsx [hunk 1]
```diff
@@ -10,14 +18,193 @@ import { Storage } from './utils/Storage'
...
+  }
+}
+
+const AppRoot: React.FC<Props> = (/* { storage } */) => {
+  const [appState, dispatchAppState] = useReducer(appReducer, initialAppState)
+  const {
+    usbDriveStatus,
+    // machineConfig,
+    electionDefinition,
+    // isTestMode,
```
Comment (benadida) [inline apps/precinct-scanner/src/AppRoot.tsx:122]: should this one not be commented out, given that you're using this appState field in later code?

**Review by benadida — APPROVED** [review]
If copying the files is the only somewhat fast path forward, I'm okay with it, but I do think we want to move these to a shared location sooner rather than later.

File: apps/precinct-scanner/src/setupProxy.js [hunk 2]
```diff
@@ -9,6 +9,8 @@
 const { createProxyMiddleware: proxy } = require('http-proxy-middleware')
 
 module.exports = function (app) {
+  app.use(proxy('/scan', { target: 'http://localhost:3002/' }))
+  app.use(proxy('/config', { target: 'http://localhost:3002/' }))
```
Comment (beausmith) [inline apps/precinct-scanner/src/setupProxy.js:13]: Should one of these ports be different?

**Review by beausmith — APPROVED** [review]

File: apps/precinct-scanner/src/setupProxy.js (hunk 2 above)
Comment (carolinemodic) [reply apps/precinct-scanner/src/setupProxy.js:13]: They're both module-scan

**Review by eventualbuddha — COMMENTED** [review]
Submitting some comments I had queued up but forgot the context on.

File: apps/precinct-scanner/package.json [hunk 3]
```diff
@@ -71,29 +71,41 @@
...
     "@testing-library/user-event": "^12.1.10",
     "@types/jest": "^24.0.0",
-    "@types/node": "^12.0.0",
+    "@types/node": "^12.20.11",
     "@types/react": "^17.0.0",
     "@types/react-dom": "^17.0.0",
     "@types/react-modal": "^3.10.5",
     "@types/react-router-dom": "^5.1.5",
+    "@votingworks/hmpb-interpreter": "workspace:*",
+    "module-scan": "workspace:*",
```
[the remark this answers is not in the document]
Comment (carolinemodic) [reply apps/precinct-scanner/package.json:80]: Ah good catch, I was playing around with seeing if I could import the types, which did not work and then forgot to remove this :)

File: apps/precinct-scanner/src/AppRoot.tsx [hunk 4]
```diff
@@ -10,14 +18,193 @@ import { Storage } from './utils/Storage'
...
+      <UnconfiguredElectionScreen
+        usbDriveStatus={usbDriveStatus}
+        setElectionDefinition={setElectionDefinition}
+      />
+    )
+  }
+
+  return (
+    <div>
+      Congratulations the precinct scanner is configured!{' '}
```
[the remark this answers is not in the document]
Comment (carolinemodic) [reply apps/precinct-scanner/src/AppRoot.tsx:204]: This is temporary text

File: apps/precinct-scanner/package.json [hunk 5]
```diff
@@ -71,29 +71,41 @@
...
     "normalize.css": "^8.0.1",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
     "react-i18next": "^11.7.3",
     "react-router-dom": "^5.2.0",
     "react-scripts": "4.0.1",
     "typescript": "^4.2.0",
-    "web-vitals": "^1.0.1"
+    "use-interval": "^1.3.0",
+    "web-vitals": "^1.0.1",
```
[the remark this answers is not in the document]
Comment (carolinemodic) [reply apps/precinct-scanner/package.json:95]: I was using use-interval? Your comment prompted me to look more at bmd though and it seems like we use @rooks/use-interval there instead? I'm not sure which is preferred but I just switched to @rooks instead.

**Pull request merged**